File: app/src/main/java/app/kamy/saatApp/domain/model/ReadingSessionModels.kt
package app.kamy.saatApp.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class ReadingSession(
    val id: String,
    val updatedAt: String? = null,
    val chapterNumber: Int,
    val verseNumber: Int
)

@Serializable
data class ReadingSessionsPage(
    val success: Boolean? = null,
    val data: List<ReadingSession>? = null,
    val pagination: CursorPagination? = null
)

@Serializable
data class CursorPagination(
    val startCursor: String? = null,
    val endCursor: String? = null,
    val hasNextPage: Boolean? = null,
    val hasPreviousPage: Boolean? = null
)

@Serializable
data class ReadingSessionInput(
    val chapterNumber: Int,
    val verseNumber: Int
)

File: iosApp/Saat/Domain/Models/ReadingSessionModels.swift
//
//  ReadingSessionModels.swift
//  Saat
//
//  Created by Elmee on 25/04/2026.
//  Copyright © 2026 Elmee. All rights reserved.
//

import Foundation

struct ReadingSession: Decodable, Sendable, Identifiable, Hashable {
    let id: String
    let updatedAt: String?
    let chapterNumber: Int
    let verseNumber: Int
}

struct ReadingSessionsPage: Decodable, Sendable {
    let success: Bool?
    let data: [ReadingSession]?
    let pagination: CursorPagination?
}

struct CursorPagination: Decodable, Sendable {
    let startCursor: String?
    let endCursor: String?
    let hasNextPage: Bool?
    let hasPreviousPage: Bool?
}
